packages/next/src/client/app-index.tsx

```diff
@@ -170,7 +172,33 @@ const initialServerResponse = createFromReadableStream(readable, {
 })
 
 function ServerRoot(): React.ReactNode {
-  return use(initialServerResponse)
+  const initialResponse = use<InitialRSCPayload>(initialServerResponse)
+
+  const {
+    b: buildId,
+    p: assetPrefix,
+    c: initialCanonicalUrl,
+    i: couldBeIntercepted,
+    t: initialTree,
+    d: initialSeedData,
+    h: initialHead,
+    m: missingSlots,
+    G: GlobalError,
+  } = initialResponse
```

Comment:
Might be worth just making AppRouter task a single prop `InitialRSCPayload` and referencing the properties by name where they are used. This might get minified OK but not really sure it helps in understanding. In fact passing it as a single object makes it a little clearer that these values aren't mutable and won't ever change for the lifetime of the App